azurerm/helpers/validate/int.go

```diff
@@ -0,0 +1,72 @@
+package validate
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// IntBetweenAndDivisibleBy returns a SchemaValidateFunc which tests if the provided value
+// is of type int and is between min and max (inclusive) and is divisible by a given number
+func IntBetweenAndDivisibleBy(min, max, divisor int) schema.SchemaValidateFunc { // nolint: unparam
+	return func(i interface{}, k string) (s []string, es []error) {
+		v, ok := i.(int)
+		if !ok {
+			es = append(es, fmt.Errorf("expected type of %s to be int", k))
+			return
+		}
+
+		if v < min || v > max {
+			es = append(es, fmt.Errorf("expected %s to be in the range (%d - %d), got %d", k, min, max, v))
+			return
+		}
+
+		if math.Mod(float64(v), float64(divisor)) != 0 {
+			es = append(es, fmt.Errorf("expected %s to be divisible by %d", k, divisor))
+			return
+		}
+
+		return s, es
+	}
+}
+
+// IntDivisibleBy returns a SchemaValidateFunc which tests if the provided value
+// is of type int and is between min and max (inclusive) and is divisible by a given number
```

Comment:
@katbyte, do you think we should update the text to the below since the function doesn't actually validate if the value is between two values?

```suggestion
// is of type int and is divisible by a given number
```